File: frontend/src/app/classes/chat-message.ts
import {User} from './user';

export class ChatMessage {

  user: User;
  message: string;
  dateSent: Date;
  dateSeen: Date;

  constructor(user: User, message: string) {
    this.user = user;
    this.message = message;
    this.dateSeen = null;
    this.dateSent = new Date();
  }
}

File: backend/src/main/java/com/fullteaching/backend/model/ChatMessage.java
package com.fullteaching.backend.model;

import lombok.*;

import javax.persistence.*;
import java.util.Date;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
@ToString
@Getter
public class ChatMessage {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;

    private String message;

    @ManyToOne(fetch= FetchType.EAGER)
    private User user;

    private Date dateSeen;

    private Date dateSent;
}
